BackEndFlask/models/user_course.py

```diff
@@ -13,23 +13,28 @@ def get_user_courses():
         error = str(e.__dict__['orig'])
         return error
 
-def get_user_courses_by_course_id(course_id):
+# Check if we also need to only return active user_courses
+def get_user_courses_by_course_id(course_id, check_active=True):
     try:
-        return UserCourse.query.filter_by(course_id=course_id).all()
+        return UserCourse.query.filter_by(course_id=course_id, active=True).all()
     except SQLAlchemyError as e:
         error = str(e.__dict__['orig'])
         return error
 
+# Update function to only return user_courses the student is active in DUN
 def get_user_courses_by_user_id(user_id):
     try:
-        return UserCourse.query.filter_by(user_id=user_id).all()
+        # return UserCourse.query.filter_by(user_id=user_id).all()
```

Comment:
No need to keep the old code in a comment. 

Additionally, there are lots of comment that seem like development notes, not documentation for future developers. I would either remove these or reword them to make the intent clear.